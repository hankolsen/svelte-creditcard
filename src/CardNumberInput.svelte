<script lang="ts">
  import handleNumberInput from './handleNumberInput';
  import isValidCreditCard from './isValidCreditCard';
  import { number } from './store';

  let error;

  const handleChange = (event) => {
    if ($number.length === 16) {
      if (!isValidCreditCard($number)) {
        error = true;
      } else {
        error = false;
      }
    }
  };
  let maxCardlength = 16;
</script>

<label for="cardNumber">Card Number</label>
<input
  name="cardNumber"
  id="cardNumber"
  bind:value={$number}
  on:keypress={handleNumberInput}
  on:keyup={handleChange}
  maxlength={maxCardlength}
  class={error ? 'card__number--error' : ''}
/>
{#if error}<p class="card__number-error-message">That seems wrong</p>{/if}

<style>
  .card__number--error,
  .card__number--error:focus {
    border-color: #b00020;
    background-color: hsl(349, 100%, 90%);
  }
  .card__number-error-message {
    padding: 0;
    text-align: right;
    margin-top: -28px;
    margin-right: 8px;
    color: #b00020;
    font-size: 0.75rem;
  }
</style>
