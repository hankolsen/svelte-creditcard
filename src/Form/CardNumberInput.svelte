<script lang="ts">
  import { number } from '@/store';
  import getCreditCardIssuer, {
    CreditCardIssuers,
  } from '@/getCardIssuer/getCardIssuer';
  import isValidCreditCard from '@/isValidCreditCard/isValidCreditCard';
  import handleNumberInput from '@/handleNumberInput/handleNumberInput';

  let error;
  let cardNumberLength;
  // prettier-ignore
  $: cardNumberLength = getCreditCardIssuer($number) === CreditCardIssuers.AMERICANEXPRESS
    ? 15
    : 16;

  const handleChange = () => {
    if ($number.length === cardNumberLength) {
      if (!isValidCreditCard($number)) {
        error = true;
      }
    } else {
      error = false;
    }
  };
</script>

<label for="cardNumber">Card Number</label>
<input
  name="cardNumber"
  id="cardNumber"
  bind:value={$number}
  on:keypress={handleNumberInput}
  on:keyup={handleChange}
  maxlength={cardNumberLength}
  class={error ? 'card__number--error' : ''}
/>
{#if error}<p class="card__number-error-message">That seems wrong</p>{/if}

<style>
  .card__number--error,
  .card__number--error:focus {
    border-color: #b00020;
    background-color: hsl(349, 100%, 90%);
  }
  .card__number-error-message {
    padding: 0;
    text-align: right;
    margin-top: -28px;
    margin-right: 8px;
    color: #b00020;
    font-size: 0.75rem;
  }
</style>
