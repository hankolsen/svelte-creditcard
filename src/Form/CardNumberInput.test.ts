/* eslint-disable import/no-extraneous-dependencies */
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/svelte';
import CardNumberInput from './CardNumberInput.svelte';

describe('CardNumberInput test', () => {
  it('should render', () => {
    const { container, getByLabelText } = render(CardNumberInput);
    expect(getByLabelText('Card Number')).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
