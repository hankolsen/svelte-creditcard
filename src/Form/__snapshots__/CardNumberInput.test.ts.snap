// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardNumberInput test should render 1`] = `
<body>
  <div>
    <label
      for="cardNumber"
    >
      Card Number
    </label>
     
    <input
      class=" svelte-1eyahzb"
      id="cardNumber"
      maxlength="16"
      name="cardNumber"
    />
     
    
  </div>
</body>
`;
